### PR TITLE
Deploy assets to S3 via GitHub Action

### PR DESCRIPTION
**Pull request by CYBAI**
With moving to GitHub Action, we can remove TravisCI for this repo.

Also see https://github.com/servo/download.servo.org/pull/75#issuecomment-823447103.

### CODE REVIEW

**Review by jdm — APPROVED** [review]

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Deploy
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
```
Comment (jdm) [inline .github/workflows/deploy.yaml]: We don't actually want this to run on pull requests, right? Only when pushing to master.

Comment (CYBAI) [reply .github/workflows/deploy.yaml]: oops, yes! Let me update. Thank you!

**Pull request merged**